release-22.1: sql: add COPY to sampled_query logging

otan opened the pull request:
Backport (combined into one commit):
  * 1/1 commits from "sql: add COPY to sampled_query logging" (#86627)
  * 1/1 commits from "sql: properly log copy error on error" (#86870)

Release justification: telemetry only change

/cc @cockroachdb/release

otan committed:
sql: add COPY to sampled_query logging

As requested by product, we want COPY to show up on sampled_query. This
commit adds this, whilst fudging a few things presumably more relevant
to "regular" DDL statements.

Release justification: telemetry only change
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87091)
<!-- Reviewable:end -->

otan commented:
blathers backport 21.2

rafiss commented:
do you know if the timing bug affects 22.1 also? should i backport https://github.com/cockroachdb/cockroach/pull/86991?

otan committed:
sql: plumb isCopy down to event log

Previously, we relied on the `txn` being filled to log `COPY`
statements. This is however inaccurate - `txn` may be nil if `COPY` is
the first statement encountered. This meant that the first COPY
encountered would not be logged.

Instead, plumb down `isCopy` and use `timeutil.Now()` if it is a COPY
related statement.

Release justification: logging related critical fix

Release note: None

otan commented:
> do you know if the timing bug affects 22.1 also? should i backport https://github.com/cockroachdb/cockroach/pull/86991?

v22.1 is fine!
